v2/client.go

```diff
@@ -31,7 +33,12 @@ var UserAgent = fmt.Sprintf("egoscale/%s (%s; %s/%s)",
 
 // defaultHTTPClient is HTTP client with retry logic.
 // Default retry configuration can be found in go-retryablehttp repo.
-var defaultHTTPClient = retryablehttp.NewClient().StandardClient()
+var defaultHTTPClient = func() *http.Client {
+	rc := retryablehttp.NewClient()
+	// silence client by default
+	rc.Logger = log.New(io.Discard, "", 0)
```

Comment:
Does it mean ALL logs will be discarded? Even the error/panic ones?

Reply:
The methods of the client should return errors

Reply:
Generic errors (eg. HTTP error code) will propagate down to egoscale same way as with standard HTTP client, but some `go-retryablehttp` specific error context is lost (eg. failed after nth retry attempt).

What we really need in egoscale is leveled logger to separate debug logs from errors. This is supported by `go-retryablehttp` (check Logger field):
https://pkg.go.dev/github.com/hashicorp/go-retryablehttp?utm_source=godoc#Client

It would be a separate effort to implement this in egoscale.